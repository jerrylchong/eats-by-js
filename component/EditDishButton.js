import React from 'react';
import {StyleSheet, Text, View, TouchableOpacity, Dimensions, Alert, AsyncStorage} from 'react-native';
import Tag from "./Tag";
import {deleteDish} from '../helpers/DishAPI'

const EditDishButton = (props) => {

    const { title, description, price, onPress, navigation, id, refresh } = props;

    return (
        <TouchableOpacity disabled={true} style = {styles.container} onPress = {onPress}>
            <View style={styles.text}>
                <Text style={styles.name}>{title}</Text>
                <Text style={styles.cost}>${price}</Text>
            </View>
            <View style={{flexDirection: 'row'}}>
                <Text style = {styles.description}>{description}</Text>
                <View style={{flexDirection: 'row', width: '20%', justifyContent: 'space-between'}}>
                    <Tag name={"Edit"} onPress={
                        () => navigation.navigate('Edit Dish',
                            {
                                id: id,
                                title: title,
                                desc: description,
                                price: price,
                            })
                    }/>
                    <Tag name={"Delete"} onPress={
                        () => {
                            Alert.alert("Delete Dish",
                                "Are you sure you want to delete " + title + "?",
                                [
                                    { text: "Cancel",
                                        onPress: () => null,
                                        style: "cancel" },
                                    { text: "Yes", onPress: () => {
                                            AsyncStorage.getItem("token")
                                                .then(token => deleteDish(id, token))
                                                .then(() => Alert.alert("Success",
                                                    "Dish " + title + " deleted."))
                                                .then(() => refresh())
                                                .catch(console.err)
                                        } }
                                ]
                            )
                        }
                    }/>
                </View>
            </View>
        </TouchableOpacity>
    )
}

export default EditDishButton

const styles = StyleSheet.create({
    container: {
        paddingHorizontal:"3%",
        width: '100%',
        height: Dimensions.get('window').height * 0.10,
        alignSelf: 'center',
        justifyContent: 'space-evenly',
        borderBottomWidth: 0.5,
        borderColor: '#b3b3b3',
        backgroundColor: 'white'
    },
    text: {
        width: '100%',
        height: '30%',
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center'
    },
    nameReview: {
        flexDirection: 'row',
        alignItems: 'center',
    },
    name: {
        color: '#404040',
        fontSize: 20,
        fontFamily: 'Ubuntu-Bold'
    },
    cost: {
        color: '#404040',
        fontSize: 20,
        fontFamily: 'Ubuntu'
    },
    description: {
        color: '#7e7e7e',
        fontSize: 11,
        alignSelf: 'flex-start',
        fontFamily: 'Ubuntu',
        width: '75%'
    },
})
