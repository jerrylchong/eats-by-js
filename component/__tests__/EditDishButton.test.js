import React from 'react';
import renderer from 'react-test-renderer';
import EditDishButton from "../EditDishButton";
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";

// snapshot test
test('renders correctly', () => {
    const tree = renderer.create(<EditDishButton title={'Title'}
                                                 description={'Desc'}
                                                 price={'5'}/>).toJSON();
    expect(tree).toMatchSnapshot();
})

Enzyme.configure({ adapter: new Adapter() });