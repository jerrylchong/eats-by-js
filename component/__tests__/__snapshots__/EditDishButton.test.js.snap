// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "white",
      "borderBottomWidth": 0.5,
      "borderColor": "#b3b3b3",
      "height": 133.4,
      "justifyContent": "space-evenly",
      "opacity": 1,
      "paddingHorizontal": "3%",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": "30%",
        "justifyContent": "space-between",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#404040",
          "fontFamily": "Ubuntu-Bold",
          "fontSize": 20,
        }
      }
    >
      Title
    </Text>
    <Text
      style={
        Object {
          "color": "#404040",
          "fontFamily": "Ubuntu",
          "fontSize": 20,
        }
      }
    >
      $
      5
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "flex-start",
          "color": "#7e7e7e",
          "fontFamily": "Ubuntu",
          "fontSize": 11,
          "width": "75%",
        }
      }
    >
      Desc
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": "20%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ff6961",
                "borderRadius": 10,
                "marginRight": 5,
              },
              Object {
                "backgroundColor": "#ff6961",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Ubuntu-Bold",
                "fontSize": 10,
                "paddingHorizontal": 9,
                "paddingVertical": 3,
              }
            }
          >
            Edit
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ff6961",
                "borderRadius": 10,
                "marginRight": 5,
              },
              Object {
                "backgroundColor": "#ff6961",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Ubuntu-Bold",
                "fontSize": 10,
                "paddingHorizontal": 9,
                "paddingVertical": 3,
              }
            }
          >
            Delete
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
